WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizer.swift

```diff
@@ -117,12 +118,16 @@ private extension RemoteOrderSynchronizer {
 
     /// Updates the underlying order as inputs are received, and triggers a remote sync for significant inputs.
     ///
-    func bindInputs() {
+    func bindInputs(flow: EditableOrderViewModel.Flow) {
         setStatus.withLatestFrom(orderPublisher)
-            .map { newStatus, order in
-                order.copy(status: newStatus)
+            .sink { [weak self] newStatus, order in
+                let orderWithNewStatus = order.copy(status: newStatus)
+                self?.order = orderWithNewStatus
+                if case .editing = flow {
+                    self?.orderSyncTrigger.send(orderWithNewStatus)
+                }
```

Comment:
nit: I find it useful to have the state mutation separate from the side effect, that way Its easy to understand what the reactive chain is doing and if its doing more than it should.

Something like
```swift
.map { newStatus, order in
    order.copy(status: newStatus)
}
.sink { [weak self] order in
    self?.order = orderWithNewStatus
    if case .editing = flow {
        self?.orderSyncTrigger.send(orderWithNewStatus)
    }
}
```

Reply:
Good point! Added in 4535167.